test/e2e/specs/site-editor/style-book.spec.js

```diff
@@ -59,37 +59,50 @@ test.describe( 'Style Book', () => {
 		).toBeVisible();
 		await expect( page.locator( 'role=tab[name="Theme"i]' ) ).toBeVisible();
 
+		// Buttons to select block examples are rendered within the Style Book iframe.
+		let styleBookIframeLocator = page.frameLocator(
+			'.edit-site-style-book__iframe'
+		);
+
 		await expect(
-			page.locator(
-				'role=button[name="Open Headings styles in Styles panel"i]'
-			)
+			styleBookIframeLocator.getByRole( 'button', {
+				name: 'Open Headings styles in Styles panel',
+			} )
 		).toBeVisible();
 		await expect(
-			page.locator(
-				'role=button[name="Open Paragraph styles in Styles panel"i]'
-			)
+			styleBookIframeLocator.getByRole( 'button', {
+				name: 'Open Paragraph styles in Styles panel',
+			} )
 		).toBeVisible();
 
 		await page.click( 'role=tab[name="Media"i]' );
 
+		// The tab has changed, which means the iframe has been re-rendered.
+		styleBookIframeLocator = page.frameLocator(
```

Comment:
If we're using `frameLocator` then we don't need to reevaluate since they will be evaluated lazily anyway. We can reuse the same locator.

Reply:
Oh, TIL, thanks Kai!

Reply:
This worked nicely!